service/orchestrator/orchestrator.go

```diff
@@ -185,6 +194,70 @@ func (s *Service) ListMetricConfigurations(ctx context.Context, req *orchestrato
 	return
 }
 
+// StoreAssessmentResult is a method implementation of the orchestrator interface: It receives an assessment result and stores it
+func (s *Service) StoreAssessmentResult(_ context.Context, req *orchestrator.StoreAssessmentResultRequest) (response *orchestrator.StoreAssessmentResultResponse, err error) {
+
+	response = &orchestrator.StoreAssessmentResultResponse{}
+
+	err = s.handleResult(req.Result)
+
+	if err != nil {
+		return response, status.Errorf(codes.InvalidArgument, "invalid assessment result: %v", err)
+	}
+
+	return
+}
+
+func (s *Service) StoreAssessmentResults(stream orchestrator.Orchestrator_StoreAssessmentResultsServer) (err error) {
+	var result *assessment.AssessmentResult
+
+	for {
+		result, err = stream.Recv()
+		if err == io.EOF {
+			log.Infof("Stopped receiving streamed assessment results")
+			return stream.SendAndClose(&emptypb.Empty{})
+		}
+
+		err = s.handleResult(result)
+		if err != nil {
+			return fmt.Errorf("error handle assessment result: %w", err)
+		}
+	}
+
+}
+
+func (s Service) handleResult(result *assessment.AssessmentResult) error {
+	_, err := result.Validate()
+	if err != nil {
+		log.Errorf("Invalid assessment result: %v", err)
```

Comment:
Minor thing: You could switch the lines 232, 233 and do sth like `log.Error(newError)` to avoid duplicated code?

Reply:
I thougt the log.Errorf() should start capitalized and the returned error lowercase.